app/models/doi.rb

```diff
@@ -2370,8 +2370,8 @@ def self.add_index_type(options = {})
 
 
   # QUICK FIX UNTIL PROJECT IS A RESOURCE_TYPE_GENERAL IN THE SCHEMA
-  def handleResourceType(types)
-    if types["resourceType"] == "Project" && (types["resourceTypeGeneral"] == "Text" || types["resourceTypeGeneral"] == "Other")
+  def handle_resource_type(types)    
+    if !types.nil? && types["resourceType"] == "Project" && (types["resourceTypeGeneral"] == "Text" || types["resourceTypeGeneral"] == "Other")
```

Comment:
This if statement could read a tiny bit better if you used `types.present?` instead of `!types.nil?`. I appreciate that this comment could be considered a nit.